src/ts/ConvivaAnalytics.ts

```diff
@@ -113,8 +166,16 @@ export class ConvivaAnalytics {
     this.sessionKey = Conviva.Constants.NO_SESSION_KEY;
     this.isAd = false;
 
-    const deviceMetadata: {[key: string]: Conviva.Constants.DeviceCategory} = {
-      [Conviva.Constants.DeviceMetadata.CATEGORY]: this.config.deviceCategory,
+    const deviceMetadataFromConfig = this.config.deviceMetadata || {};
+    const deviceMetadata = {
```

Comment:
Is it recommended by Conviva to always send all of the `deviceMetadata` fields, even if they might be `undefined`?

Reply:
In practice, for Conviva library, sending them as undefined is the same that not sending them at all. We need to define in this way because the user can provide those values or not.